pkg/gatewayserver/gatewayserver.go

```diff
@@ -654,18 +654,17 @@ func (gs *GatewayServer) handleUpstream(conn connectionEntry) {
 	}
 }
 
-// UpdateConnectionStats updates the stats for a single gateway connection.
-func (gs *GatewayServer) UpdateConnectionStats(ctx context.Context, conn *io.Connection) error {
-	return gs.statsRegistry.Set(ctx, conn.Gateway().GatewayIdentifiers, conn.Stats())
-}
-
 func (gs *GatewayServer) updateConnStats(conn connectionEntry) {
 	ctx := conn.Context()
 	logger := log.FromContext(ctx)
 
+	if gs.statsRegistry == nil {
+		panic("No stats registry available")
+	}
+
 	defer func() {
 		logger.Debug("Delete connection stats")
-		err := gs.statsRegistry.Set(ctx, conn.Gateway().GatewayIdentifiers, nil)
+		err := conn.ClearStats(gs.statsRegistry)
```

Comment:
Are the statistics cleared even if upstream/downstream/status can be connected still to another instance?

Reply:
Initially did not account for this, I have updated the code to only clear the specified statistics